src/components/Header/Header.tsx

```diff
@@ -65,6 +67,11 @@ const Header: React.FC<IHeader> = ({ title, debounceSearch }) => {
     handlePopState();
   }, [location]);
 
+  useEffect(() => {
+    const timer = isFlipping ? setTimeout(() => setIsFlipping(false), 500) : undefined;
+    return () => clearTimeout(timer);
+  }, [isFlipping, setIsFlipping]);
```

Comment:
@vinaybadgujar102 please correct me if i m wrong
Including setIsFlipping in the dependency array might seem redundant in this context. When setIsFlipping is called, it inherently triggers a change in the isFlipping state right ?

Reply:
agreed, including "setIsFlipping" is redundant